lib/Echidna/SymExec.hs

```diff
@@ -0,0 +1,56 @@
+module Echidna.SymExec where
+
+import Control.Monad (forM)
+import Control.Monad.State.Strict (evalStateT)
+import Data.Map qualified as Map
+import Data.Maybe (catMaybes, fromMaybe)
+
+import EVM (StorageModel(..))
+import EVM.Expr (simplify)
+import EVM.Fetch qualified as Fetch
+import EVM.Solidity (SolcContract(..))
+import EVM.Solvers (withSolvers, Solver(Z3), CheckSatResult(Sat))
+import EVM.SymExec (interpret, runExpr, abstractVM, mkCalldata, produceModels)
+import EVM.Types
+import EVM.SMT
+
+import Echidna.Types.Tx
+
+exploreContract :: Addr -> SolcContract -> IO [Tx]
+exploreContract dst contract = do
+  let
+    calldata = mkCalldata Nothing [] -- fully abstract calldata, solution in buffers.txdata
+    -- Alternatively, more concrete calldata with a specific function,
+    -- not sure what is better yet
```

Comment:
I think you're probably going to make life easier for the solvers if you use more specialized calldata, as it allows us to simplify away quite a lot of abi decoding (amongst other things)